src/v/datalake/translation/partition_translator.cc

```diff
@@ -77,6 +77,20 @@ ss::futurize_t<FuncRet> retry_with_backoff(
     }
 }
 
+static std::unique_ptr<record_translator>
+make_record_translator(model::iceberg_mode mode) {
+    switch (mode) {
+    case model::iceberg_mode::disabled:
+        vassert(
+          false,
+          "Cannot make record translator when iceberg is disabled, logic bug.");
+    case model::iceberg_mode::key_value:
+        return std::make_unique<key_value_translator>();
+    case model::iceberg_mode::value_schema_id_prefix:
+        return std::make_unique<default_translator>();
```

Comment:
Should be structured_data_translator?

Reply:
doesn't the default translator fall back to kv incase of resolution issues or I missed something there?

```
record_type
default_translator::build_type(std::optional<resolved_type> val_type) {
    if (val_type.has_value()) {
        return structured_translator.build_type(std::move(val_type));
    }
    return kv_translator.build_type(std::move(val_type));
}
```

Reply:
It does, but default_translator is a hack to smooth out the transition. Resolution issues should be disallowed (dropped) when using the schema_id_prefix mode IIUC

Reply:
Agreed - no kv fallback was the agreement with Matt from the schemas doc